File: src/go/power_float.go
package main

import (
	"fmt"
	"math"
	"os"
	"strconv"
	"time"
)

func main() {
	base := 2.0
	exp := 1023

	if baseStr := os.Getenv("BASE"); baseStr != "" {
		if val, err := strconv.ParseFloat(baseStr, 64); err == nil {
			base = val
		}
	}

	if expStr := os.Getenv("EXP"); expStr != "" {
		if val, err := strconv.Atoi(expStr); err == nil {
			exp = val
		}
	}

	start := time.Now()
	// Using standard library math.Pow for fairness in "native" float comparison
	result := math.Pow(base, float64(exp))
	elapsed := time.Since(start)

	fmt.Printf("Result(%.2f^%d): %.10e\n", base, exp, result)
	fmt.Printf("Time: %.3f ms\n", float64(elapsed.Nanoseconds())/1000000.0)
}

File: src/rust/power_float.rs
use std::env;
use std::time::Instant;

fn main() {
    let base: f64 = env::var("BASE")
        .unwrap_or("2".to_string())
        .parse()
        .unwrap_or(2.0);
    
    let exp: i32 = env::var("EXP")
        .unwrap_or("1023".to_string())
        .parse()
        .unwrap_or(1023);

    let start = Instant::now();
    let result = base.powi(exp); // powi is for integer exponent, native f64 method
    let duration = start.elapsed();

    println!("Result({}^{}): {:e}", base, exp, result);
    println!("Time: {:.3} ms", duration.as_nanos() as f64 / 1_000_000.0);
}
